### PR TITLE
fix(components/colorpicker): icon color accessibility account for alpha

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[AB#2619788](https://blackbaud.visualstudio.com/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2619788)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1511/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1511/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1511/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1511/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1511/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1511/playground/)

File: libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts [hunk 1]
```diff
@@ -619,16 +619,36 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
...
+      const G =
+        GsRGB <= 0.03928
+          ? GsRGB / 12.92
+          : Math.pow((GsRGB + 0.055) / 1.055, 2.4);
+      const B =
+        BsRGB <= 0.03928
+          ? BsRGB / 12.92
+          : Math.pow((BsRGB + 0.055) / 1.055, 2.4);
+
+      const relativeLuminence =
```
Comment (johnhwhite) [inline libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts]: Check the spelling.

File: libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts [hunk 2]
```diff
@@ -619,16 +619,36 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
...
+        (0.2126 * R + 0.7152 * G + 0.0722 * B) *
+        (1 / backgroundColor.rgba.alpha);
```
Comment (johnhwhite) [inline libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts:646]: This gets weird if `backgroundColor.rgba.alpha` is `0` — can we return `undefined` in that case so it just uses the body text color?

Comment (johnhwhite) [reply libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts:646]: `1/0` resolves to black, which works with the checkered backdrop we currently have.

**Review by johnhwhite — APPROVED** [review]

**Pull request merged**